File: core/java/src/main/java/hara/lang/protocol/IConj.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.iconj", name = "IConj")
public interface IConj<E> {
  @HaraMethod(value = "conj", arity = 2)
  IConj<E> conj(E e);
}

File: core/rust/src/lang/protocol/iconj.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.iconj", name = "IConj")]
pub trait IConj<E> {
    type Output;

    #[hara_method(value = "conj", arity = 2)]
    fn conj(&self, value: E) -> Self::Output;
}
